### PR TITLE
Validate the URL is unique

### PR DESCRIPTION
**Pull request by rimian**
This checks a record with the same URL does not exist with a different ID.

### CODE REVIEW

File: lib/validate_url.rb [hunk 1]
```diff
@@ -53,18 +54,25 @@ def validate_url(record, attribute, value, schemes)
...
+        unless valid_raw_url && valid_scheme && valid_no_local && valid_suffix && valid_uniq
           record.errors.add(attribute, options.fetch(:message), value: value)
         end
       rescue URI::InvalidURIError
         record.errors.add(attribute, :url, **filtered_options(value))
       end
+
+      # Validates the URL is unique without the schema
+      def url_uniq?(record, attribute, url)
+        record.class.where("#{attribute} LIKE ?", "%#{url}").where.not(id: record.id).size.zero?
```
Comment (rimian) [inline lib/validate_url.rb:74]: Is there a better way to interpolate this? I tried many things...

File: README.md [hunk 2]
```diff
@@ -42,6 +42,9 @@ class Pony < ActiveRecord::Base
 
   # with Postgres array of urls, described [here](https://guides.rubyonrails.org/active_record_postgresql.html#array)
   validates :homepage, url: { accept_array: true }
+
+  # Validates the URL is unique without the scheme part
+  validates :homepage, url: { uniq_url: true }
```
Comment (rimian) [inline README.md]: I think I will change this to `unique_url`

File: lib/validate_url.rb (hunk 1 above)
Comment (willbuilds) [reply lib/validate_url.rb:74]: nah I wouldn't worry, it looks fine, does what we need :)

Comment (willbuilds) [reply lib/validate_url.rb:74]: Alternative could be interpolation with named params, but I think it's more wordy for the sake of it... yours is clearer.
```
record.class.where(":attribute LIKE :url", attribute: attribute, url: "%#{url}")
```

File: spec/validate_url_spec.rb [hunk 3]
```diff
@@ -345,4 +345,18 @@
       expect(user.errors[:homepage]).to eq(['wrong'])
     end
   end
+
+  context 'with unique URL' do
+    let!(:user) { UserWithUniqUrl.new(homepage: 'https://example.test').save(validate: false) }
+
+    it 'is not valid if the schema is different' do
```
Comment (willbuilds) [inline spec/validate_url_spec.rb:352]: maybe add a spec where it's identical as well (same schema)

**Review by willbuilds — APPROVED** [review]
looks good mate, nothing major, make some changes if you think it's helpful

**Pull request merged**